include/modem/location.h

```diff
@@ -0,0 +1,343 @@
+ *
+ * @param[in] event_data Event data.
+ */
+typedef void (*location_event_handler_t)(const struct location_event_data *event_data);
+
+/**
+ * @brief Initializes the library.
+ *
+ * @details Initializes the library and sets the event handler function.
+ * This can be called multiple times, which sets the event handler again.
+ *
+ * @param[in] event_handler Event handler function.
+ *
+ * @return 0 on success, or negative error code on failure.
+ * @retval -EINVAL Given event_handler is NULL.
+ * @retval -EFAULT Failed to obtain Wi-Fi interface.
+ */
+int location_init(location_event_handler_t event_handler);
+
+/**
+ * @brief Requests the current position or starts periodic position updates.
+ *
+ * @details Requests the current position using the given configuration. Depending on the
+ * configuration, a single position or periodic position updates are given. If the position
+ * request is successful, the results are delivered to the event handler function in
+ * LOCATION_EVT_LOCATION event. Otherwise LOCATION_EVT_TIMEOUT or LOCATION_EVT_ERROR event is
+ * triggered.
+ *
+ * In periodic mode, position update is always attempted after the configured interval
+ * regardless of the outcome of any previous attempt. Periodic position updates can be
+ * stopped by calling location_cancel().
+ *
+ * @param[in] config Used configuration or NULL to get a single position update with
+ *                   the default configuration.
+ *
+ * @return 0 on success, or negative error code on failure.
+ * @retval -EPERM  Library not initialized.
+ * @retval -EBUSY  Previous location request still ongoing.
+ * @retval -EINVAL Invalid configuration given.
+ */
+int location_request(const struct location_config *config);
+
+/**
+ * @brief Cancels periodic position updates.
+ *
+ * @return 0 on success, or negative error code on failure.
+ * @retval -EPERM  Library not initialized.
+ */
+int location_request_cancel(void);
+
+/**
+ * @brief Sets the default values to a given configuration.
+ *
+ * @details Given methods are set as a list of methods for the given configuration.
+ * Caller is responsible for allocating and deallocating memory for both configuration and
+ * method list. The 'methods' memory area must be kept allocated as long as the configuration
+ * is used. Using a statically allocated list is encouraged.
+ *
+ * @param[in,out] config Configuration to be supplied with default values.
+ * @param[in] methods_count Number of methods to be stored in 'methods'.
+ * @param[in] methods List of methods. This will be initialized to zero.
+ */
+void location_config_defaults_set(
+	struct location_config *config,
+	uint8_t methods_count,
+	struct location_method_config *methods);
+
+/**
+ * @brief Sets the default values for a given method configuration based on the given type.
+ *
+ * @details This is part of location_config that has just been initialized, meaning that
+ * this function is called right after location_config_defaults_set().
+ *
```

Comment:
Why is this a separate function, if it always needs to be called when after `location_config_defaults_set()`?

Reply:
location_config_defaults_set() haven't had information on what location methods to initialize to the methods list.
However, this has been refactored significantly and this function is no longer in the API.